components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java

```diff
@@ -0,0 +1,260 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.siddhi.metrics.core;
+
+import org.apache.log4j.Logger;
+import org.testng.Assert;
+import org.testng.AssertJUnit;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+import org.wso2.carbon.metrics.core.Level;
+import org.wso2.carbon.metrics.core.MetricManagementService;
+import org.wso2.carbon.metrics.core.MetricService;
+import org.wso2.carbon.metrics.core.Metrics;
+import org.wso2.carbon.siddhi.metrics.core.internal.SiddhiMetricsDataHolder;
+import org.wso2.carbon.siddhi.metrics.core.internal.SiddhiStatisticsManager;
+import org.wso2.siddhi.core.SiddhiAppRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.config.StatisticsConfiguration;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.stream.output.StreamCallback;
+import org.wso2.siddhi.core.util.EventPrinter;
+import org.wso2.siddhi.core.util.statistics.EventBufferHolder;
+
+/**
+ * Test case for carbon metrics inside siddhi.
+ */
+public class StatisticsTestCase {
+    private static final Logger log = Logger.getLogger(StatisticsTestCase.class);
+    private int count;
+    private boolean eventArrived;
+    protected static Metrics metrics;
+
+    protected static MetricService metricService;
+
+    protected static MetricManagementService metricManagementService;
+
+    @BeforeSuite
+    protected void initMetrics() throws Exception {
+        // Initialize the Metrics
+        System.setProperty("metrics.target", "target");
+        metrics = new Metrics(TestUtils.getConfigProvider("metrics.yaml"));
+        metrics.activate();
+        metricService = metrics.getMetricService();
+        metricManagementService = metrics.getMetricManagementService();
+    }
+
+    @BeforeMethod
+    public void init() {
+        metricManagementService.setRootLevel(Level.ALL);
+        metricManagementService.stopReporters();
+        count = 0;
+        eventArrived = false;
+        SiddhiMetricsDataHolder.getInstance().setMetricService(metricService);
+        SiddhiMetricsDataHolder.getInstance().setMetricManagementService(metricManagementService);
+        metricManagementService.startReporter("Console");
+    }
+
+    @AfterSuite
+    protected static void destroy() throws Exception {
+        metrics.deactivate();
+    }
```

Comment:
Add new line here

Reply:
Fixed in b4a2bcf444d25031775a679bad76798d70ce7d5b